File: backend/TransportesOrellanaSpa.Api/Models/Remolque.cs
namespace TransportesOrellanaSpa.Api.Models;

public class Remolque
{
    public int Id { get; set; }

    public string Patente { get; set; } = string.Empty;
    public string Marca { get; set; } = string.Empty;
    public string Modelo { get; set; } = string.Empty;

    public int Ano { get; set; }

    public string Tipo { get; set; } = string.Empty;

    public double CapacidadToneladas { get; set; }

    public bool Activa { get; set; } = true;

    // Tractocamión habitual del Remolque.
    public int? CamionHabitualId { get; set; }
    public Camion? CamionHabitual { get; set; }

    public ICollection<Viaje> Viajes { get; set; } = new List<Viaje>();
}

File: frontend/TransportesOrellanaSpa.Web/src/app/core/models/remolque.ts
import { CamionResumen } from './camion-resumen';

export interface Remolque {
  id: number;
  patente: string;
  marca: string;
  modelo: string;
  ano: number;
  tipo: string;
  capacidadToneladas: number;
  activa: boolean;
  camionHabitualId: number | null;
  camionHabitual: CamionResumen | null;
}